### PR TITLE
Bump node to 18

### PR DESCRIPTION
**Pull request by HarlemSquirrel**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     "settings": {
         "import/parsers": {
             "@typescript-eslint/parser": [".ts", ".tsx"]
-        }
+        },
+        "jest": { "version": 27 }
```
Comment (HarlemSquirrel) [inline .eslintrc.json:34]: We need this and `eslint-jest-plugin` because `eslint-config-preact` requires it :expressionless: 
https://github.com/preactjs/eslint-config-preact/issues/19#issuecomment-997924892

**Comment by HarlemSquirrel** [comment]
Was able to build and install locally but so far unable to connect to KeeWeb from extension when built from this branch.

**Comment by HarlemSquirrel** [comment]
Well for some reason tonight when I rebuilt it now it's working :sweat_smile:

**Comment by Aetherinox** [comment]
Got to love when that happens.
FYI, as of today's changes, ensure your environment uses only one of the following NodeJS versions:
- `^18.12.0`
- `^20.9.0`
- `>=22.0.0`

If KeeWeb is built on any other version; there will be dependency errors and things will break.

Is this PR completely done now?

**Comment by HarlemSquirrel** [comment]
I'd like to test the chrome extension but haven't yet. That's it

**Comment by HarlemSquirrel** [comment]
Having trouble connecting to Chrome with local built extension. I think it may be due to the extension ID being different but updating https://github.com/keeweb/keeweb/blob/master/desktop/scripts/const/extension-ids.js and running keeweb locally doesn't solve it for me.

**Comment by HarlemSquirrel** [comment]
Still not working with node 16 or 18 on master branch, last tag, or this branch. I'm clearly missing something.

**Comment by HarlemSquirrel** [comment]
@antelle sorry to bug you but do you have any thoughts on what I might be missing here running chrome/chromium extension locally and connecting to keeweb desktop? Firefox works but in Chromium I get no logs in the desktop console which leads me to believe it's not receiving any messages at all. The background extension console says `KeeWeb host disconnected Specified native messaging host not found.`

**Comment by antelle** [comment]
Do you have the messaging host config enabled? When you check the checkbox in settings in KeeWeb, it creates the config, but I assume it has a wrong extension ID there

**Comment by HarlemSquirrel** [comment]
OK I'm not sure why but I got it working on a different computer running Pop OS 22.04 so I'm going to say this is working OK with Firefox and Chrome.

**Comment by HarlemSquirrel** [comment]
I also successfully tested edge

**Comment by Aetherinox** [comment]
Did you get all of it working now? I still need to mess with this to get a feel for it. But I'll have to do it after I finish up the stuff with KeeWeb. Don't want to start flip-flopping.

**Comment by HarlemSquirrel** [comment]
Yeah it's working locally. There does appear to be a validation issue when trying to publish the Firefox addon but I have a separate PR for that

**Comment by Aetherinox** [comment]
NIce, progress. I'll definitely look into these plugins after I wrap up my end. I spend half my time reading pages of code so that I know what all is done in that segment, trying to get myself familiarized with this app like I would my own. Mainly prepping myself for the Electron v14+ migration since a lot has changed, and I don't want ti keep us running on v13 for too long.
